File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s831281137_comments_removed.py
import sys

def main():
    N, W = map(int, sys.stdin.readline().split())
    v = [0] * (N + 1)
    w = [0] * (N + 1)
    for i in range(1, N+1):
        vi, wi = map(int, sys.stdin.readline().split())
        v[i] = vi
        w[i] = wi
    sumV = [[0]*(W+1) for _ in range(N+1)]
    for i in range(1, N+1):
        current_w = w[i]
        for s in range(current_w):
            sumV[i][s] = sumV[i-1][s]
        for s in range(current_w, W+1):
            opt1 = sumV[i-1][s]
            opt2 = sumV[i-1][s - current_w] + v[i]
            sumV[i][s] = max(opt1, opt2)
    print(sumV[N][W])

if __name__ == "__main__":
    main() # End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C++/s831281137_comments_removed.cpp
#include <iostream>
#include <vector>
using namespace std;

int nint() {
    int x;
    cin >> x;
    return x;
}

int main() {
    int N = nint();
    int W = nint();

    vector<int> v(N+1);
    vector<int> w(N+1);

    for (int i = 1; i <= N; i++) {
        v[i] = nint();
        w[i] = nint();
    }

    vector<vector<int>> sumV(N+1, vector<int>(W+1));

    for (int i = 1; i <= N; i++) {
        for (int s = 0; s < w[i]; s++) {
            sumV[i][s] = sumV[i-1][s];
        }
        for (int s = w[i]; s <= W; s++) {
            sumV[i][s] = max(sumV[i-1][s], sumV[i-1][s - w[i]] + v[i]);
        }
    }

    cout << sumV[N][W] << endl;
    return 0;
} // End of Code
